exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Table.kt

```diff
@@ -1179,6 +1181,17 @@ open class Table(name: String = "") : ColumnSet(), DdlAware {
         getRealColumns().filter { !it.columnType.nullable }.forEach { (it as Column<Any>).nullable() }
     } as CompositeColumn<T?>
 
+    /**
+     * Appends a database-specific column [definition] to this column's SQL in a CREATE TABLE statement.
+     *
+     * The specified [definition] is appended after the column's name, type, and default value (if any),
+     * but before any column constraint definitions. If multiple definition arguments are passed, they
+     * will be joined as string representations separated by a single space character.
+     */
+    fun <T> Column<T>.withDefinition(vararg definition: Any): Column<T> = apply {
+        extraDefinitions.addAll(definition)
+    }
```

Comment:
Suggestions are welcome both for **names** and function **parameter**.

I also considered: `withOption()`, `useDefinition()`, `addDefinition()`, `definition()`.

**Parameter alternatives:**

`Column<T>.withDefinition(vararg definition: Expression<*>)`
- Attempting to pass in a string as `stringLiteral()` would wrap it in single quotes, which is appropriate in query clauses, but not in create DDL.

`Column<T>.withDefinition(definition: String)`
- Attempting something like `.withDefinition("ON UPDATE $CurrentTimestamp")`, would cause `Please call Database.connect() before using this code` because it is trying to resolve the expression to string outside of transaction.

2 overloads, so the 2 options above together
- But I thought it looked more busy: `.withDefinition("ON UPDATE").withDefinition(CurrentTimestamp)`

Reply:
`withDefinition` looks good